api/chain.go

```diff
@@ -461,6 +471,59 @@ func (a *API) chainTxByIndexHandler(msg *apirest.APIdata, ctx *httprouter.HTTPCo
 	return ctx.Send(data, apirest.HTTPstatusOK)
 }
 
+// chainTxByHeightHandler
+//
+//	@Summary		Returns the list of transactions for a given block
+//	@Description	Given a block returns the list of transactions for that block
+//	@Success		200	{object}	TransactionList
+//
+//	@Router			/chain/transactions/reference/height/{height} [get]
+func (a *API) chainTxByHeightHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	height, err := strconv.ParseUint(ctx.URLParam("height"), 10, 64)
+	if err != nil {
+		return err
+	}
+	block := a.vocapp.GetBlockByHeight(int64(height))
+	if block == nil {
+		return ErrBlockNotFound
+	}
+	blockTxs := &BlockTransactionsInfo{
+		BlockNumber:       height,
+		TransactionsCount: uint32(len(block.Txs)),
+		Transactions:      make([]TransactionMetadata, len(block.Txs)),
+	}
+	for i := range block.Txs {
+		signedTx := new(models.SignedTx)
+		tx := new(models.Tx)
+		var err error
+		if err = proto.Unmarshal(block.Txs[i], signedTx); err != nil {
+			return ErrUnmarshalingServerProto.WithErr(err)
+		}
+		if err = proto.Unmarshal(signedTx.Tx, tx); err != nil {
+			return ErrUnmarshalingServerProto.WithErr(err)
+		}
+		txType := string(
+			tx.ProtoReflect().WhichOneof(
+				tx.ProtoReflect().Descriptor().Oneofs().Get(0)).Name())
+
+		txRef, err := a.indexer.GetTxHashReference(block.Txs[i].Hash())
+		if err != nil {
+			return ErrTransactionNotFound
+		}
+		blockTxs.Transactions = append(blockTxs.Transactions, TransactionMetadata{
+			Type:   txType,
+			Index:  int32(i),
+			Height: uint32(txRef.Index),
```

Comment:
index and height are both indices? confusing :)

Reply:
yes it is a bit confusing, but there is the need for changing this names in multiple places. For now we agree that index is the tx index in the block, and height is the tx index in the overall chain.